src/main/java/dev/dbos/transact/DBOS.java

```diff
@@ -326,7 +344,7 @@ public void send(String destinationId, Object message, String topic) {
    * @param timeoutSeconds time in seconds after which the call times out
    * @return the message if there is one or else null
    */
-  public Object recv(String topic, float timeoutSeconds) {
+  public Object recv(String topic, double timeoutSeconds) {
```

Comment:
We should be using Duration for these timeout parameters. Optionally, also provide overloads that take long+TimeUnit params and construct a Duration from Duration.fromNanos